server/middleware/featureGate.ts

```diff
@@ -1,9 +1,32 @@
 import type { NextFunction, Request, Response, RequestHandler } from 'express'
 import { NotFound } from 'http-errors'
 
-export default function featureGate(flag: string, handler: RequestHandler): RequestHandler {
+import { Caseload } from '../data/nomisUserRolesApi'
+
+/**
+ * Wraps a request handler and returns 404 unless the given feature flag is set
+ */
+export function featureGate(flag: string, handler: RequestHandler): RequestHandler {
+  return (req: Request, res: Response, next: NextFunction): void => {
+    if (req.app.locals?.featureFlags?.[flag]) {
+      handler(req, res, next)
+    } else {
+      next(new NotFound())
+    }
+  }
+}
+
+/**
+ * Wraps a request handler and returns 404 unless any of the user’s case loads appears in given prisons
+ */
+export function caseloadGate(prisons: string[], handler: RequestHandler): RequestHandler {
   return (req: Request, res: Response, next: NextFunction): void => {
-    if (req.app.locals.featureFlags[flag]) {
+    const activeCaseload: Caseload | undefined = res.locals?.user?.activeCaseload
+    const caseloads: Caseload[] = res.locals?.user?.caseloads ?? []
+    if (
+      (activeCaseload && prisons.includes(activeCaseload.id)) ||
+      caseloads.map(caseload => caseload.id).some(prison => prisons.includes(prison))
```

Comment:
I think this middleware should only check for the active case load and ignore the other caseloads a user may have (maybe the name of the middleware should reflect this? e.g. `activeCaseLoadGate`? 🤷)

Assuming the handler uses the active case load to do something with it (e.g. [filter data/show a graph for the active case load prison](https://github.com/ministryofjustice/hmpps-incentives-ui/pull/85/commits/2646bb775b7ef619c56547e3c17146ba757e6d5e#diff-46a3cef6cd40be5b2c529ed431954c6e9b197b0218eed6ef43ddbf99b7286972R33)).

Is this to potentially avoid confusion for users with some case loads where analytics are shown but not in all?

Reply:
that's in exactly what i have in a subsequent commit! i thought it'd be more readable to have it in separate ones